### DISCUSSION TITLE
Overlay one gif on top of another

### DISCUSSION THREAD

**Discussion by onigiri-x** [ImageMagick/ImageMagick#6740] [Digital Image Processing]
Hi there

I have a series of gifs I am trying to create and blend together.

So I have a background gif. This gif has a green background, and has some animation on top of that

then i have the character gif that goes on top of that. The gif has transparency. Then in the middle is the body of the character. Then there is animation on top of that character (shiny skin)

So I have one gif background and another gif to overlay directly on top. both have animations. both have 10 frames. And also, there are other gif backgrounds that have no animation that also have to work, and other characters with no animation as well.

Im using the command

magick convert ./Traits/Background/GreenAnimatedBackground.gif null: ./Traits/Character/ShinyCharacter.gif +dither -layers composite result69.gif

The result basically overlays both the animation from the background layer and the animation from the foreground layer on top of everything. so although i have my gif character overlayed on top of the green animated background, I still see the animation from the green animated background in the forefront meshed with the shiny animation.

Any help would be extremely awesome right now as I have a tight deadline on it. Didnt think magick would be difficult to use.

Thanks

**Comment by snibgo** [comment]
You have IM v7, but use `magick convert` which uses the old (very old) v6 syntax. I suggest using just `magick` instead.

I don't understand what problem you need to solve.

If an animated GIFs is compressed, you will need `-layers coalesce` after reading it.

If you have an animated GIF but want only the first frame, suffix the filename with `[0]`.

I suggest you post example input GIFs, and explain what result you want.

**Comment by fmw42** [comment]
Please post the two example animations and your resulting animation after applying your command above.
